### DISCUSSION TITLE
Add percentile normalization

### DISCUSSION THREAD

**Discussion by timonmerk** [neuromodulation/py_neuromodulation#370] [Ideas]
An option could be added in the nm_settings.json s.t. `raw_normalization_settings` / `feature_normalization_settings`  have an argument `use_percentile`, `lower_percentile`, `upper_percentile` which limits the mean/median/z-score data distribution and addresses the effect of outliers. 

See discussion in #285

**Comment by toni-neurosc** [comment]
Hi Timon, I was taking a look at those RuntimeErrors that were appearing in the normalization and I remembered this issue. Is this still interesting? It's easy to implement, but I'm not sure if do it as you say in "normalization_settings" or have it as an extra preprocessing step that you can add to the "preprocessing" list. 

Also, in the discussion #285 you said that `Commonly we face the problem that there are movement/cable noise artifacts that exceed many standard deviations of the data`. If the artifacts deviate in many sigmas, then shouldn't the "clip" parameter of the normalization settings take care of that? 

As long as you have z-score as the normalization method, the clipping is done in terms of standard deviations, the problem is what happens you're using any of the other normalization methods, or you're using no normalization at all, which seems to be the case you described in your message. In this case, it would be kinda useless to put the `use_percentile` option in the normalization settings, if you still skip normalization.

For this reason, I propose adding it as an extra normalization step, like `remove_outliers` or `clip_percentiles`, whatever you think is more descriptive. It's either that, or add a new dummy normalization method `none` that does not modify the data, so that you can use normalization with a `use_percentile` parameter without actually normalizing, but i feel it's too convoluted.
